src/css.js

```diff
@@ -283,9 +284,9 @@ jQuery.extend({
 				return;
 			}
 
-			// If a number was passed in, add 'px' (except for certain CSS properties)
+			// If a number was passed in, add the unit (except for certain CSS properties)
 			if ( type === "number" && !jQuery.cssNumber[ origName ] ) {
-				value += "px";
+				value += ret && ret[ 3 ] ? ret[ 3 ] : "px";
```

Comment:
Optimization opportunity: `value += ret && ret[ 3 ] || "px"` (or, if `rrelNum` is no more, `ret && ret[ 1 ] && ret[ 3 ] || "px"`).